Action log

tom-MO opened the pull request:
[Jira Task](https://myoxygenltd.atlassian.net/browse/MYOX-7?atlOrigin=eyJpIjoiZTBhMDJhMjRjNWUyNGE2NzgwZjZkNGYzODEwZTViZWEiLCJwIjoiaiJ9)

This PR looks at implementing `ActionLog` (previously partially implemented in ScholarPack) and an associate log file viewer to allow test users to navigate through the logs. It gives a good indication of what's going on in the app, and allows testers to see what is coming and going from the servers.

The actual widgets allow the user to copy and delete log files as they wish. These files can also be deleted from the device's app settings menu (by doing "clear data/cache"), but visually seening this list of files is clearer to understand, and they are not deleted when they are copied (unlike the ScholarPack implementation).

There is an example project that demonstrates its behaviour and the intended implementation.

tom-MO committed:
[ActionLog] Initial commit

tom-MO committed:
[ActionLog] Addit folder name to CircleCI config

tom-MO committed:
[ActionLog] Added dependencies and first code

tom-MO committed:
[ActionLog] Created paginated UI for displaying logs

tom-MO committed:
[ActionLog] Created basic UI for displaying logs

tom-MO committed:
[ActionLog] Created and implemented example project

tom-MO committed:
[ActionLog] Updated files and dependencies

tom-MO committed:
[ActionLog] Added comments to classes and functions

tom-MO committed:
[ActionLog] Changed the colour of the arrow to show the item is active

tom-MO committed:
[ActionLog] Added copy and delete actions to log file viewer

tom-MO committed:
[ActionLog] Added delete button to the logs list.

tom-MO committed:
[ActionLog] Removed redundant functions

tom-MO committed:
[ActionLog] Synchronised the log messages

Ensures that only one log message is written to the log file at a time.

tom-MO committed:
[ActionLog] Fixed crashing bug

This would happen when the user would navigate back from the log file viewer.

tom-MO committed:
ActionLog] Added tests for ActionLogHelper

tom-MO committed:
[ActionLog] Added some crude widget tests

Not really confident about these tests. They mainly fail because the ListView isn't loading, despite having called tester.pump() or tester.pumpAndSettle(). Who knows why...

tom-MO committed:
[ActionLog] If custom file name is used, display that instead.

tom-MO committed:
[ActionLog] Updated ReadMe

MO-James commented on action_log/.idea/workspace.xml:
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
A lot of these files are IntelliJ/AS specific. 
Can you add them to .gitignore if appropriate. 
This one should be ignored https://stackoverflow.com/questions/27647889/should-i-add-idea-workspace-xml-to-gitignore

MO-James commented on action_log/CHANGELOG.md:
```diff
@@ -0,0 +1,3 @@
+## 0.0.1
+
+* TODO: Describe initial release.
```
Leftover (autogenerated?) todo

MO-James commented on action_log/LICENSE:
```diff
@@ -0,0 +1 @@
+TODO: Add your license here.
```
Leftover (autogenerated?) todo

MO-James commented on action_log/README.md:
````diff
@@ -0,0 +1,78 @@
+    ref: ActionLog-v0.0.1 # Use the latest ActionLog tag!!
+```
+
+### SnackBarPresenter's APIs
````
This is `ActionLog` not `SnackbarPresenter`

MO-James reviewed (commented):
Code looks fine for a quick look over. 
Can you remove the `.gitignore`able files? It's hard to do a code review on 84 files when most aren't important.

tom-MO committed:
Minor tweaks to project files.

tom-MO commented:
@MO-James Addressed your comments. I've removed the `.idea` directory, but that still leaves about 80 files. Most of these are native project files, which are required to build and run. Were there any other specific files/folders you noticed?

MO-James reviewed (approved):
Cool, LGTM
